core/data/src/main/java/uk/gov/gchq/gaffer/data/element/Element.java

```diff
@@ -42,7 +42,7 @@
  * <p>
  * Equals has been overridden to check groups are equal. NOTE - it does not compare property values.
  */
-@JsonTypeInfo(use = Id.CLASS, include = As.EXISTING_PROPERTY, property = "class")
+@JsonTypeInfo(use = Id.CLASS, include = As.PROPERTY, property = "class")
```

Comment:
I realise the reason for changing it from EXTERNAL_PROPERTY, as EXTERNAL_PROPERTY doesn't work with databind in the older version of jackson.

However, changing to PROPERTY may cause duplicate "class" properties in the JSON.

Due to serialising and deserialising generic types we had an issue where the "class" field was being completely missed off. To ensure it is always present we added a getter and setter for "class". The EXISTING_PROPERTY annotation then just relies on the getter and setter being used.